File: iRHSCBook/RHSCMyBookingsList.m
//
//  RHSCMyBookingsList.m
//  iRHSCBook
//
//  Created by Bruce Hunter on 2014-07-08.
//  Copyright (c) 2014 Richmond Hill Squash Club. All rights reserved.
//

#import "RHSCMyBookingsList.h"
#import "RHSCTabBarController.h"
#import "RHSCCourtTime.h"
#import "RHSCUser.h"

@implementation RHSCMyBookingsList

- (void)loadFromJSON:(RHSCServer *)server user:(RHSCUser *)curUser {
    // Create a NSURLRequest with the given URL
    NSString *fetchURL = [NSString stringWithFormat:@"Reserve20/IOSMyBookingsJSON.php?uid=%@",curUser.data.name];

//    NSLog(@"fetch URL = %@",fetchURL);
    
    NSURL *target = [[NSURL alloc] initWithString:fetchURL relativeToURL:server];
    NSURLRequest *request = [NSURLRequest requestWithURL:[target absoluteURL]
                                             cachePolicy:NSURLRequestReloadIgnoringLocalAndRemoteCacheData
                                         timeoutInterval:30.0];
	
    // Get the data
    NSURLResponse *response;
	NSData *data = [NSURLConnection sendSynchronousRequest:request returningResponse:&response error:nil];
    
    // Now create a NSDictionary from the JSON data
    [self loadFromData:data forUser:curUser.data.name];
}

- (void)loadFromData:(NSData *) data forUser:(NSString *)userId {
    // Create a NSURLRequest with the given URL
    // Now create a NSDictionary from the JSON data
    NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
    
    // Create a new array to hold the locations
    NSMutableArray *bookList = [[NSMutableArray alloc] init];
    
    // Get an array of dictionaries with the key "locations"
    NSArray *array = [jsonDictionary objectForKey:@"bookings"];
    // Iterate through the array of dictionaries
    for(NSDictionary *dict in array) {
        // Create a new Location object for each one and initialise it with information in the dictionary
        RHSCCourtTime *booking = [[RHSCCourtTime alloc] initWithJSONDictionary:dict forUser:userId];
        // Add the Location object to the array
        [bookList addObject:booking];
    }
    _bookingList = [[NSArray alloc] initWithArray:bookList];
}

@end

File: iRHSCBook/RHSCMyBookingsList.swift
//
//  RHSCMyBookingsList.swift
//  iRHSCBook
//
//  Created by Bruce Hunter on 2015-10-25.
//  Copyright © 2015 Bruce Hunter. All rights reserved.
//

import Foundation
import UIKit

@objc class RHSCMyBookingsList : NSObject {

    var list = Array<RHSCCourtTime>()

    func loadFromJSON(fromServer server:RHSCServer, user curUser:RHSCUser, memberList ml:RHSCMemberList) throws {
        let url = URL(string: String.init(format: "Reserve20/IOSMyBookingsJSON.php?uid=%@",curUser.name!),
            relativeTo: server as URL )
        //        print(url!.absoluteString)
        let sessionCfg = URLSession.shared.configuration
        sessionCfg.timeoutIntervalForResource = 30.0
        let session = URLSession(configuration: sessionCfg)
        let task = session.dataTask(with: url!, completionHandler: { (data, response, error) -> Void in
            if error != nil {
                print("Error: \(error!.localizedDescription)")
            } else if data != nil {
                //                print(NSString(data: data!, encoding: NSUTF8StringEncoding))
                self.loadFromData(data!, forUser: (curUser.name)!, memberList: ml)
            }
        })
        task.resume()
    }
        func loadFromData(_ fromData:Data, forUser: String, memberList ml:RHSCMemberList) {
        do {
            if let jsonDictionary = try JSONSerialization.jsonObject(with: fromData, options: []) as? NSDictionary {
                let array : Array<[String : String]> = jsonDictionary["bookings"]! as! Array<[String : String]>
                for dict in array {
                    list.append(RHSCCourtTime(withJSONDictionary: dict, forUser: forUser, members: ml))
                }
            }
        } catch {
            print(error)
        }
    }

    func loadAsync(fromView view: UITableViewController) {
        let tbc = view.tabBarController as! RHSCTabBarController
        let url = URL(string: String.init(format: "Reserve20/IOSMyBookingsJSON.php?uid=%@",
            (tbc.currentUser?.name)!),
            relativeTo: tbc.server as URL? )
        //        print(url!.absoluteString)
        //        let sessionCfg = NSURLSession.sharedSession().configuration
        //        sessionCfg.timeoutIntervalForResource = 30.0
        //        let session = NSURLSession(configuration: sessionCfg)
        let session  = URLSession.shared
        let task = session.dataTask(with: url!, completionHandler: { (data, response, error) -> Void in
            if error != nil {
                print("Error: \(error!.localizedDescription)")
            } else if data != nil {
                //                print(NSString(data: data!, encoding: NSUTF8StringEncoding))
                self.loadFromData(data!, forUser: tbc.currentUser!.name!, memberList: tbc.memberList!)
            }
            DispatchQueue.global().async(execute: {
                // do some task
                DispatchQueue.main.async(execute: {
                    // update some UI
                    view.tableView.reloadData()
                });
            });
        })
        task.resume()
    }
    

}
